cuda/base/stream.cpp

```diff
@@ -30,58 +30,47 @@ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ******************************<GINKGO LICENSE>*******************************/
 
-#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/stream.hpp>
 
 
-#include <thread>
+#include <cuda_runtime.h>
 
 
-#include <gtest/gtest.h>
+#include <ginkgo/core/base/exception_helpers.hpp>
 
 
-namespace {
+#include "cuda/base/scoped_device_id.hpp"
 
 
-#define GTEST_ASSERT_NO_EXIT(statement) \
-    ASSERT_EXIT({ {statement} exit(0); }, ::testing::ExitedWithCode(0), "")
+namespace gko {
 
 
-TEST(DeviceReset, HipCuda)
-{
-    GTEST_ASSERT_NO_EXIT({
-        auto ref = gko::ReferenceExecutor::create();
-        auto hip = gko::HipExecutor::create(0, ref, true);
-        auto cuda = gko::CudaExecutor::create(0, ref, true);
-    });
-}
+cuda_stream::cuda_stream() : stream_{}, device_id_{-1} {}
```

Comment:
```suggestion
cuda_stream::cuda_stream() : stream_{nullptr}, device_id_{-1} {}
```
I'm not sure about the `-1`, since passing the nullptr as a stream is valid and results in the default stream getting used. So perhaps a valid device id would be appropriate.

Reply:
agree, the device_id doesn't play an important role except for the destruction, and we don't expose it otherwise